tests/product/timing_test_decorator.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import sys
+
+from re import match
+from time import time
+
+
+def _initialize_module_logger(module_name):
+    logger = logging.getLogger(module_name)
+    logger.setLevel(logging.INFO)
+
+    message_format = '%(name)s - %(message)s'
+    formatter = logging.Formatter(message_format)
+
+    console_handler = logging.StreamHandler(sys.stdout)
+    console_handler.setLevel(logging.INFO)
+    console_handler.setFormatter(formatter)
+
+    if not len(logger.handlers):
+        logger.addHandler(console_handler)
+    return logger
+
+
+def _name_starts_with_test(name):
+    pattern = '^test.*$'
+    return match(pattern, name)
+
+
+# Logging is disabled by default for product tests
+# Logging can be enabled using this module by uncommenting the lines:
+# logging.disable(logging.NOTSET) in the function below
+def log_function_time(module_name="root"):
+    """
+    Args:
+        module_name: The module of the function that is being timed
+         By default, this is set to 'root' to use the root logger.
+         It is recommended that __name__ be passed into this function
+         for the most accurate logging.
+
+    Usage:
+         @log_function_time(__name__)
+         def my_function(params):
+
+    Returns: Prints the execution time of the decorated function to the
+    log file(s). It will always log as info the execution time to the
+    info log file. If the execution time exceeds 10 minutes, it will also
+    log it to the error log file.
+    """
+    def name_wrapper(function):
+        def time_wrapper(*args, **kwargs):
+            # logging.disable(logging.NOTSET)
+            logger = _initialize_module_logger(module_name)
```

Comment:
It looks like logging is initialized every time this decorator is used. It would be better if we found a way to initialize the logging module once only. This race condition, for lack of a better term, is why you have an `if` statement on line 19, to avoid adding a handler every time logging is setup. Unfortunately, I don't how to do that so I'm gonna have to think about it. @ebd2 do you have any ideas?